pkg/domain/nexpose/asset.go

```diff
@@ -0,0 +1,272 @@
+package nexpose
+
+import (
+	"os"
+)
+
+// Asset represents a Nexpose asset response payload
```

Comment:
This statement means that these types are _not_ part of the project domain but are, instead, part of the specific HTTP implementation of `AssetFetcher` that this project provides.

That being said, I'm open to arguments that the only _real_ domain of this project _is_ the nexpose HTTP API. If so then we may make an exception to our usual rule of `no JSON annotation in domain`. For the common case, having JSON annotations on a domain type indicates that we don't have a domain abstraction and any change in the remote API that produces/consumes these will require changes to _all_ consumers of the project domain. This is exactly the problem that having a `domain` package is intended to mitigate.

Reply:
> ...the only real domain of this project is the nexpose HTTP API

I think this is true. We're trying to provide everything that's provided by Nexpose to our consumers and I'd expect to have to make changes to some of our services if the Nexpose API changes